### PR TITLE
Fix CI/CD

### PR DESCRIPTION
**Pull request by Lucien950**
### Summary

Change up the Clion run configs.
Major TODO is to add support for the new symbol files. There does not seem to be support for GDB config settings in Clion, which means that we manually add symbol files?? Or maybe it auto detects it. I am not sure. @gtaharaedmonds

### Changelist 

- [x] Fix damage from #1145 
    - [x] Change name of deployed binaries
    - [x] Change name of CLion embedded build targets
- [x] Run DBC CD only when requested
   - [x] Delete Release task now needs to check if there was a release made in the first place
- [x] Single Line Release Deploy now Multiline 
- [x] Check for Protoc to Build System (Mandatory)
- [x] Delete all old tags

### Resolved Issues

- Fixes [SW-54](https://ubcformulaelectric.atlassian.net/jira/software/projects/SW/boards/20?selectedIssue=SW-54) and [SW-60](https://ubcformulaelectric.atlassian.net/jira/software/projects/SW/boards/20?selectedIssue=SW-60)

### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
- [x] I have read and followed the code conventions detailed in [README.md](../README.md) (*This will save time for both you and the reviewer!*).
- [x] If this pull request is longer then **500** lines, I have provided *explicit* justification in the summary above explaining why I *cannot* break this up into multiple pull requests (*Small PR's are faster and less painful for everyone involved!*).

[SW-54]: https://ubcformulaelectric.atlassian.net/browse/SW-54?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[SW-60]: https://ubcformulaelectric.atlassian.net/browse/SW-60?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/actions/push_release/action.yml
Comment (gtaharaedmonds) [inline .github/actions/push_release/action.yml:30]: `debug env` and `debug name` aren't very descriptive, could pick a better name

File: firmware/boot/config.h
Comment (gtaharaedmonds) [inline firmware/boot/config.h:72]: These CAN IDs need to be different than the VCs, this is how each bootloader is uniquely identified on the bus. 
The convention I've been running with is 11xx IDs are for Thruna, 12xx IDs are for Quadruna, and 13xx IDs are for dev boards. And then a range of 10 is reserved for each board in the correct 100 range. Also fixed the pin by looking at Altium schematics.

```
#define START_UPDATE_ID 1200
#define UPDATE_ACK_ID 1201
#define STATUS_10HZ_ID 1209
#define BOOT_GPIO_PORT GPIOC
#define BOOT_GPIO_PIN GPIO_PIN_13
```

Comment (gtaharaedmonds) [reply firmware/boot/config.h:72]: Can you also add this to in `can_bus/quadruna/Boot` so this shows up in the DBC? (reference the VC for how to do it)

File: firmware/quadruna/BMS/CMakeLists.txt
Comment (gtaharaedmonds) [inline firmware/quadruna/BMS/CMakeLists.txt:134]: Looks like you need to change the linker script from `*app_only.ld` to `*app.ld`
You should test this on the BMS before merging.

File: scripts/utilities/generate_log4j_properties.py
Comment (gtaharaedmonds) [inline scripts/utilities/generate_log4j_properties.py:30]: Now that we're using this again, how do we use it?

Comment (gtaharaedmonds) [reply scripts/utilities/generate_log4j_properties.py:30]: I can test it on Mac

**Review by gtaharaedmonds — CHANGES_REQUESTED** [review]
Various comments

File: scripts/utilities/generate_log4j_properties.py
Comment (Lucien950) [reply scripts/utilities/generate_log4j_properties.py:30]: this just runs at generate time

**Comment by gtaharaedmonds** [comment]
lots of good cleanup tests here, thanks

File: scripts/utilities/generate_log4j_properties.py
Comment (Lucien950) [reply scripts/utilities/generate_log4j_properties.py:30]: it will create a specific config file that stm looks for in your directory structure

File: firmware/boot/config.h
Comment (Lucien950) [reply firmware/boot/config.h:72]: is it boot_fdcan 1 or 2?

File: firmware/quadruna/BMS/CMakeLists.txt
Comment (Lucien950) [reply firmware/quadruna/BMS/CMakeLists.txt:134]: ok test.

File: firmware/boot/config.h
Comment (gtaharaedmonds) [reply firmware/boot/config.h]: FDCAN1
Whats the error?

Comment (Lucien950) [reply firmware/boot/config.h]: https://github.com/UBCFormulaElectric/Consolidated-Firmware/blob/e011f4572e47f1699bf69ea79c56c96212fe5d86/firmware/boot/h7boot/cubemx/Src/main.c#L154

**Review by gtaharaedmonds — APPROVED** [review]
ok lgtm now feel free to merge when my comments are all addressed

File: can_bus/quadruna/Boot/Boot_tx.json
Comment (Lucien950) [inline can_bus/quadruna/Boot/Boot_tx.json:1]: jk, just checked the blame i ain't never done that

**Pull request merged**
